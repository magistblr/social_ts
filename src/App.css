@import '/src/fonts.css';

* {
  font-family: 'Proxima Nova';
}


.app-wrapper {
  display: block;
  margin: 0 300px 0 300px;
  width: 1100px;
}

.app-wrapper-inner {
  display: grid;
  grid-template-areas:
            'header header'
            'navbar content';
  grid-gap: 10px;
  grid-template-columns: 250px 100%;
  grid-template-rows: 70px 1fr;

}

.item {
  padding: 10px;
}


