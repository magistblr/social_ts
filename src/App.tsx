import React from 'react';
import { Route } from 'react-router-dom';
import s from './App.module.css';

import News from './components/News/News';
import Music from './components/Music/Music';
import Settings from './components/Settings/Settings';
import SidebarContainer from './components/Sidebar/SidebarContainer';
import UsersContainer from './components/Users/UsersContainer';
import ProfileContainer from './components/Profile/ProfileContainer';
import HeaderContainer from './components/Header/HeaderContainer';
import {Login} from './components/Login/Login';
import DialogsContainer from './components/Dialogs/DialogsContainer';


const App = () => {
  return (
    <div className={s.app_wrapper}>
      <div className={s.app_outer}>
        <div className={s.app_inner}>
            <HeaderContainer />
          <div className={s.wrapper}>
            <SidebarContainer />
            <div className={s.content}>
              <Route path="/profile/:userId?" render={() => <ProfileContainer/>}/>
              <Route path="/messages" render={ () => <DialogsContainer />}/>
              <Route path="/users" render={ () => <UsersContainer />}/>
              <Route path="/news" render={() => <News />} />
              <Route path="/music" render={() => <Music />} />
              <Route path="/settings" render={() => <Settings />} />
              <Route path="/login" render={ () => <Login/>}/>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default App;
