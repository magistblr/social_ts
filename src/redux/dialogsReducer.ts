import { v1 } from 'uuid';
import { FriendMessageType } from '../components/Dialogs/Message/FriendMessage';
import { MessageType } from '../components/Dialogs/Message/Message';


export type DialogsPageType = {
  dialogs: Array<DialogsType>
  messages: Array<MessageType>
  friendMessages: Array<FriendMessageType>
  newMessageBody: string
}

export type DialogsType = {
  id: string
  name: string
}


export type DialogsActionTypes = ReturnType<typeof sendMessageCreator> | ReturnType<typeof updateMessageCreator>


const NEW_MESSAGE = "NEW_MESSAGE";
const SEND_MESSAGE = "SEND_MESSAGE";

let initialState: DialogsPageType = {
  dialogs: [
    {id: v1(), name: 'Dima'},
    {id: v1(), name: 'Andrey'},
    {id: v1(), name: 'Sveta'},
    {id: v1(), name: 'Sasha'},
    {id: v1(), name: 'Victor'},
    {id: v1(), name: 'Valera'}
  ],
  messages: [
    {id: v1(), message: 'Hi'},
  ],
  friendMessages: [
    {id: v1(), message: 'How are you?'}
  ],
  newMessageBody: ''
}


const dialogsReducer = (state = initialState, action: DialogsActionTypes): DialogsPageType => {
    switch (action.type) {
      case NEW_MESSAGE:
        state.newMessageBody = action.newText
        let text: MessageType = {id: v1(), message: action.newText}
        state.messages.push(text)
        return state;
      case SEND_MESSAGE:
        state.newMessageBody = action.newText
        return state;
      default:
        return state;
    }
}


export const sendMessageCreator  = (text: string) => ({type: NEW_MESSAGE, newText: text} as const)
export const updateMessageCreator  = (text: string) => ({type: SEND_MESSAGE, newText: text} as const)


export default dialogsReducer;
