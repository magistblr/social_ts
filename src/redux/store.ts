

export type StoreType = {
  _state: StateType
  addPost: (postText: string) => void
  onPostChange: (newText: string) => void
  subscriber: (renderTree: () => void) => void
  _render: () => void
  getState: () => StateType
}

export type StateType = {
  profilePage: ProfilePageType
  dialogsPage: DialogsPageType
  sidebarPage: SidebarPageType
  friendsBar:  FriendsBarType
}

export type ProfilePageType = {
  posts: Array<PostsType>
  newPostText: string
  profile: null
}

export type PostsType = {
  id: number
  message: string
}

export type DialogsPageType = {
  dialogs: Array<DialogsType>
  messages: Array<MessageType>
  friendMessages: Array<FriendMessagesType>
  newMessageBody: string
}

export type DialogsType = {
  id: number
  name: string
}

export type MessageType = {
  id: number
  message: string
}

export type NavbarType = {
  id: number
  name: string
  url: string
}

export type FriendMessagesType = {
  id: number
  message: string
}

export type FriendsType = {
  id: number
  name: string
}

export type FriendsBarType = {
  friends: Array<FriendsType>
}

export type SidebarPageType = {
  navbar: Array<NavbarType>
}



let store: StoreType = {
  _state:  {

    profilePage:  {
      posts:  [
        {id: 1, message: 'Hi, how are you?'},
        {id: 2, message: "It's my first post"}
      ],
      newPostText: '',
      profile: null
    },

    dialogsPage: {
      dialogs:  [
        {id: 1, name: 'Dima'},
        {id: 2, name: 'Andrey'},
        {id: 3, name: 'Sveta'},
        {id: 4, name: 'Sasha'},
        {id: 5, name: 'Victor'},
        {id: 6, name: 'Valera'}
      ],
      messages: [
        {id: 1, message: 'Hi'},
      ],
      friendMessages: [
        {id: 1, message: 'How are you?'}
      ],
      newMessageBody: ''
    },

    sidebarPage: {
      navbar: [
        {id: 1, name: 'Profile', url: '/profile'},
        {id: 2, name: 'Messages', url: '/messages'},
        {id: 3, name: 'News', url: '/news'},
        {id: 4, name: 'Music', url: '/music'},
        {id: 5, name: 'Settings', url: '/settings'},
      ]
    },

    friendsBar: {
      friends: [
        {id: 1, name: 'Andrew'},
        {id: 2, name: 'Sasha'},
        {id: 3, name: 'Sveta'},
      ]
    },
  },

  _render() {
    console.log(("state changed"))
  },

  getState() {
    return (this._state);
  },

  addPost(postText: string) {
    const newPost: PostsType = {
      id: new Date().getTime(),
      message: postText
    }
    this._state.profilePage.posts.push(newPost)
    this._render();
  },

  onPostChange(newText: string) {
    this._state.profilePage.newPostText = newText
    console.log("render");
    this._render()
  },

  subscriber(renderTree: () => void) {
    this._render = renderTree;
  }
}



//позволяет выводить state в консоли
//@ts-ignore
window.store = store;

export default store;




