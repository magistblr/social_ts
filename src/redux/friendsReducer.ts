import { v1 } from 'uuid';
import { ActionTypes, FriendsBarType } from './store';


// const NEW_MESSAGE = "NEW_MESSAGE";
// const SEND_MESSAGE = "SEND_MESSAGE";

let initialState: FriendsBarType = {
      friends: [
        {id: v1(), name: 'Andrew'},
        {id: v1(), name: 'Sasha'},
        {id: v1(), name: 'Sveta'},
      ]
}


const friendsReducer = (state = initialState, action: ActionTypes) => {
    // switch (action.type) {
    //   case NEW_MESSAGE:
    //     state.newMessageBody = action.newText
    //     let text: MessageType = {id: v1(), message: 'Hi'}
    //     state.messages.push(text)
    //     return state;
    //   case SEND_MESSAGE:
    //     state.newMessageBody = action.newText
    //     return state;
    //   default:
        return state;
    }


// export const sendMessageCreator  = (text: string) => ({type: NEW_MESSAGE, newText: text})
// export const updateMessageCreator  = (text: string) => ({type: SEND_MESSAGE, newText: text})


export default friendsReducer;
