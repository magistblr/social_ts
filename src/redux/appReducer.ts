import { UsersType } from '../components/Users/Users';
import { authAPI, ResultCodesEnum } from "../api/api";
import { Dispatch } from "redux";
import { ThunkAction } from 'redux-thunk';
import { StateType } from './store';
import { getAuthUserData } from './authReducer';

const SET_INITIALIZED = 'SET_INITIALIZED';


export type AppPropsType = {
  initialized: boolean,
}

export type AppActionTypes = ReturnType<typeof initializedSuccess>

let initialState: AppPropsType = {
  initialized: false,
};

const appReducer = (state = initialState, action: AppActionTypes): AppPropsType => {
  switch (action.type) {
    case SET_INITIALIZED:
      return {
        ...state,
        initialized: true,
      }
    default:
      return state;
  }
}


export const initializedSuccess = () => ({type: SET_INITIALIZED} as const)

//thunk-logic
type ThunkType = ThunkAction<void, StateType, unknown, AppActionTypes>

export const initialize = (): ThunkType => (dispatch) => {
  let promise = dispatch(getAuthUserData())
  Promise.all([promise])
    .then(() => {
    dispatch(initializedSuccess())
  })
}


export default appReducer;