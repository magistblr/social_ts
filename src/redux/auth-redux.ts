import { authAPI } from "../api/api";
import { Dispatch } from "redux";

const SET_USER_DATA = 'SET_USER_DATA';


export type AuthPropsType = {
  users: any
  email: any
  login: any
  isAuth: boolean
}

export type AuthActionTypes = ReturnType<typeof setAuthUserData>

let initialState = {
  users: null,
  email: null,
  login: null,
  isAuth: false
};

const authReducer = (state = initialState, action: AuthActionTypes) => {
  switch (action.type) {
    case SET_USER_DATA:
      return {
        ...state,
        ...action.data,
        isAuth: true
      }
    default:
      return state;
  }
}

export const setAuthUserData = (userId: number, email: string, login: string) => ({type: SET_USER_DATA, data: {userId, email, login}} as const)
export const getAuthUserData = () => (dispatch: Dispatch) => {
  authAPI.me().then(response => {
    if (response.data.resultCode === 0) {
      let {id, login, email} = response.data.data;
      dispatch(setAuthUserData(id, email, login));
    }
  });
}


export default authReducer;