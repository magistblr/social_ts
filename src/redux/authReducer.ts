import { authAPI, ResultCodesEnum } from "../api/api";
import { Dispatch } from "redux";
import { ThunkAction } from 'redux-thunk';
import { StateType } from './store';

const SET_USER_DATA = 'SET_USER_DATA';
const SET_MESSAGE_ERROR_DATA = 'SET_MESSAGE_ERROR_DATA';


export type AuthPropsType = {
  userId: string
  email: string
  login: string
  isAuth: boolean
  message: string
}

export type AuthActionTypes = ReturnType<typeof setAuthUserData> |
                              ReturnType<typeof setErrorMessageData>

let initialState: AuthPropsType = {
  email: "",
  login: "",
  isAuth: false,
  userId: "",
  message: ""
};

const authReducer = (state = initialState, action: AuthActionTypes): AuthPropsType => {
  switch (action.type) {
    case SET_USER_DATA:
      return {
        ...state,
        ...action.data,
      }
      case SET_MESSAGE_ERROR_DATA:
        debugger
        return {
          ...state,
          message: action.payload.message
        }
    default:
      return state;
  }
}


export const setAuthUserData = (userId: string, email: string, login: string, isAuth: boolean) => ({type: SET_USER_DATA, data: {userId, email, login, isAuth}} as const)
export const setErrorMessageData = (message:string) => ({type: SET_MESSAGE_ERROR_DATA, payload: {message}} as const)

//thunk-logic
type ThunkType = ThunkAction<void, StateType, unknown, AuthActionTypes>

export const getAuthUserData = (): ThunkType => (dispatch: Dispatch<AuthActionTypes>) => {
  authAPI.me().then(response => {
    if (response.data.resultCode === 0) {
      let {id, login, email} = response.data.data;
      dispatch(setAuthUserData(id, email, login, true));
    }
  });
}

export const login = (email: string, password: string, remember: boolean): ThunkType => (dispatch) => {
  authAPI.login(email, password, remember).then(response => {
    if (response.resultCode === ResultCodesEnum.Success) {
      dispatch(getAuthUserData());
    }
  });
}

export const error = (): ThunkType => (dispatch) => {
  authAPI.me().then(response => {
    if (response.data.resultCode === ResultCodesEnum.Error) {
      dispatch(setErrorMessageData(response.data.message[0]));
    }
  });
}

export const logout = (): ThunkType => (dispatch) => {
  authAPI.logout().then(response => {
    if (response.resultCode === ResultCodesEnum.Success) {
      dispatch(setAuthUserData("", "", "", false));
    }
  });
}


export default authReducer;