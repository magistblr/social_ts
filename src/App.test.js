import React from 'react';
import {render, screen} from "@testing-library/react"
import App from './App';
import { Provider } from 'react-redux';
import { store } from './redux/store';
import { HashRouter } from 'react-router-dom';


  test('should ', () => {
    const fragment =
      <HashRouter>
        <Provider store={store}>
            <App />
        </Provider>
      </HashRouter>
  
    const {asFragment} = render(fragment)
    // const linkElement = getByText(/News/i);
    expect(asFragment(fragment)).toMatchSnapshot()
    // screen.debug();
  })

