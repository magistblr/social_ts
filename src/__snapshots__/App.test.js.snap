// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should  1`] = `
<DocumentFragment>
  <div
    class="app_wrapper"
  >
    <div
      class="app_outer"
    >
      <div
        class="app_inner"
      >
        <header
          class="header"
        >
          <img
            alt="logo"
            class="header_img"
            src="soc_logo.png"
          />
          <div
            class="header_login"
          >
            <a
              href="#/login"
            >
              Login
            </a>
          </div>
        </header>
        <div
          class="wrapper"
        >
          <nav
            class="wrapper"
          >
            <div>
              <div
                class="list"
              >
                <div
                  class="list_items"
                >
                  <a
                    class="item"
                    href="#/profile"
                  >
                    Profile
                  </a>
                </div>
              </div>
            </div>
            <div>
              <div
                class="list"
              >
                <div
                  class="list_items"
                >
                  <a
                    class="item"
                    href="#/messages"
                  >
                    Messages
                  </a>
                </div>
              </div>
            </div>
            <div>
              <div
                class="list"
              >
                <div
                  class="list_items"
                >
                  <a
                    class="item"
                    href="#/users"
                  >
                    Users
                  </a>
                </div>
              </div>
            </div>
            <div>
              <div
                class="list"
              >
                <div
                  class="list_items"
                >
                  <a
                    class="item"
                    href="#/news"
                  >
                    News
                  </a>
                </div>
              </div>
            </div>
            <div>
              <div
                class="list"
              >
                <div
                  class="list_items"
                >
                  <a
                    class="item"
                    href="#/music"
                  >
                    Music
                  </a>
                </div>
              </div>
            </div>
            <div>
              <div
                class="list"
              >
                <div
                  class="list_items"
                >
                  <a
                    class="item"
                    href="#/settings"
                  >
                    Settings
                  </a>
                </div>
              </div>
            </div>
          </nav>
          <div
            class="content"
          >
            <div
              class="wrapper"
            >
              <div
                class="login"
              >
                <form
                  class="form"
                >
                  <input
                    class="input"
                    name="email"
                    placeholder="email"
                    type="email"
                  />
                  <div>
                     
                  </div>
                  <input
                    class="input"
                    name="password"
                    placeholder="password"
                    type="password"
                  />
                  <div>
                     
                  </div>
                  <div
                    class="checkbox_wrapper"
                  >
                    <input
                      class="checkbox"
                      name="rememberMe"
                      type="checkbox"
                    />
                    <span
                      class="text"
                    >
                      remember me
                    </span>
                  </div>
                  <div>
                     
                  </div>
                  <input
                    class="button"
                    type="submit"
                    value="LOGIN"
                  />
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
